pkg/admiralctl/migrate/migrate.go

```diff
@@ -0,0 +1,694 @@
+
+	if o.Cluster == "" {
+		return fmt.Errorf("the cluster cannot be empty")
+	}
+
+	if o.ClusterKubeConfig == "" {
+		return fmt.Errorf("the clusterKubeConfig cannot be empty")
+	}
+
+	if o.OutputFormat != "" && o.OutputFormat != "yaml" && o.OutputFormat != "json" {
+		return fmt.Errorf("output format is only one of json and yaml")
+	}
+
+	return nil
+}
+
+// Run migrate resources from legacy clusters
+func (o *CommandMigrateOption) Run(f util.Factory, args []string) error {
+	var memberClusterFactory cmdutil.Factory
+	var err error
+
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
+	kubeConfigFlags.KubeConfig = &o.ClusterKubeConfig
+	kubeConfigFlags.Context = &o.ClusterContext
+	memberClusterFactory = cmdutil.NewFactory(kubeConfigFlags)
+
+	objInfo, err := o.getObjInfo(memberClusterFactory, o.Cluster, args)
+	if err != nil {
+		return fmt.Errorf("failed to get resource in cluster(%s). err: %w", o.Cluster, err)
+	}
+
+	obj := objInfo.Info.Object.(*unstructured.Unstructured)
+
+	o.gvk = obj.GetObjectKind().GroupVersionKind()
+
+	controlPlaneRestConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return fmt.Errorf("failed to get control plane rest config. err: %w", err)
+	}
+
+	mapper, err := restmapper.MapperProvider(controlPlaneRestConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create restmapper: %w", err)
+	}
+
+	gvr, err := restmapper.GetGroupVersionResource(mapper, o.gvk)
+	if err != nil {
+		return fmt.Errorf("failed to get gvr from %q: %w", o.gvk, err)
+	}
+
+	return o.migrate(controlPlaneRestConfig, obj, gvr)
+}
+
+func (o *CommandMigrateOption) migrate(controlPlaneRestConfig *rest.Config, obj *unstructured.Unstructured,
+	gvr schema.GroupVersionResource,
+) error {
+	if err := preprocessResource(obj); err != nil {
+		return fmt.Errorf("failed to preprocess resource %q(%s/%s) in control plane: %w", gvr, o.Namespace, o.name, err)
+	}
+
+	if o.OutputFormat != "" {
+		// only print the resource template and Policy
+		err := o.printObjectAndPolicy(obj)
+
+		return err
+	}
+
+	if o.DryRun {
+		return nil
+	}
+
+	controlPlaneDynamicClient := dynamic.NewForConfigOrDie(controlPlaneRestConfig)
+
+	kubeadmiralClient := fedclient.NewForConfigOrDie(controlPlaneRestConfig)
+
+	if len(obj.GetNamespace()) == 0 {
+		_, err := controlPlaneDynamicClient.Resource(gvr).Get(context.TODO(), o.name, metav1.GetOptions{})
+		if err == nil {
+			fmt.Printf("Resource %q(%s) already exist in Kubeadmiral control plane, "+
+				"you can edit PropagationPolicy and OverridePolicy to propagate it\n", gvr, o.name)
+			return nil
+		}
+
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get resource %q(%s) in control plane: %w", gvr, o.name, err)
+		}
+
+		_, err = controlPlaneDynamicClient.Resource(gvr).Create(context.TODO(), obj, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to create resource %q(%s) in control plane: %w", gvr, o.name, err)
+		}
+
+		if o.AutoCreatePolicy {
```

Comment:
We can create propagation policy first, then we can specify the policy when creating resource. and we don't need one more patch